### DISCUSSION TITLE
Solving ImportError: version `GLIBCXX_3.4.26' not found in Debian 10

### DISCUSSION THREAD

**Discussion by jasperhyp** [rdkit/rdkit#4758] [Q&A] [answered]
The information about my issue is here: https://github.com/rdkit/rdkit/issues/4756.

It is indeed because the conda env is using the OS libstdc++.so file rather than the conda one.  By using `find / -name "libstdc++.so.6*"`, I found a whole lot of libstdc++ files...
![image](https://user-images.githubusercontent.com/52191793/144279331-0f602fe4-026a-44c7-86b5-7d65224f2ca4.png)

I currently worked it around by resetting the system-wise dynamic link `/lib/x86_64-linux-gpu/libstdc++.so.6` pointing to the `libstdc++.so.6.0.29` in the conda env, but I do hope to find a solution to correctly set up my conda env and set the system-wise link back to normal.  Would appreciate your help!

**Comment by ptosco** [comment, marked as answer]
You really should not do that - that will break many, many things.
The correct `RPATH` should be baked into `cDataStructs.so`, as that's how `conda` works.
```
$ patchelf --print-rpath /opt/conda/envs/chemprop/lib/python3.8/site-packages/rdkit/DataStructs/cDataStructs.so
$ORIGIN/../../../..
```
As a workaround,
```
export LD_LIBRARY_PATH=/opt/conda/envs/chemprop/lib:$LD_LIBRARY_PATH
```
before running your Python script. But again, in a healthy `conda` env you do not need that.

**Reply by jasperhyp** [reply]
Thanks Paolo for the suggestion and workaround!  I am removing the redundant dynamic libraries now, and will set up a new conda environment, maybe in a Docker later.  I just can't figured out why my conda env is not healthy since I just set it up this morning using the scripts provided by [chemprop](https://github.com/chemprop/chemprop) without adding anything...

**Reply by jasperhyp** [reply]
It looks like this is a known [issue](https://github.com/chemprop/chemprop/issues/228) of Chemprop, but they currently do not have solutions.  I would stick to your solution for now.
